tests/doc_test/doc_needextend_dynamic/conf.py

```diff
@@ -0,0 +1,13 @@
+extensions = ["sphinx_needs"]
+needs_build_json = True
+
+
+def get_matching_need_ids(app, need, needs, id_prefix=""):
+    filtered_needs = []
+    for need_id, _ in needs.items():
+        if id_prefix and need_id.startswith(id_prefix):
+            filtered_needs.append(need_id)
+    return filtered_needs
+
+
+needs_functions = [get_matching_need_ids]
```

Comment:
should be added with the sphinx API, as in practical use cases (to avoid messing up the incremental build)